extensions/integration-tests/src/schemaCompare.test.ts

```diff
@@ -251,11 +255,78 @@ class SchemaCompareTester {
+		assert(t2Difference !== undefined, 'The difference Table t2 should be found. Should not be undefined');
+		const excludeResult = await schemaCompareService.schemaCompareIncludeExcludeNode(operationId, t2Difference, false, azdata.TaskExecutionMode.execute);
+		this.assertIncludeExcludeResult(excludeResult, false, 1, 0);
+		assert(excludeResult.blockingDependencies[0].sourceValue[1] === 'v1', `Blocking dependency should be view v1. Actual: ${excludeResult.blockingDependencies[0].sourceValue[1]}`);
+
+		// Exclude the view v1 that t2 was a dependency for and it should succeed and t2 should also be excluded
+		const v1Difference = schemaCompareResult.differences.find(e => e.sourceValue && e.sourceValue[1] === 'v1' && e.name === 'SqlView');
+		assert(v1Difference !== undefined, 'The difference View v1 should be found. Should not be undefined');
+		const excludeResult2 = await schemaCompareService.schemaCompareIncludeExcludeNode(operationId, v1Difference, false, azdata.TaskExecutionMode.execute);
+		this.assertIncludeExcludeResult(excludeResult2, true, 0, 1);
+		assert(excludeResult2.affectedDependencies[0].sourceValue[1] === 't2', `Table t2 should be the affected dependency. Actual: ${excludeResult2.affectedDependencies[0].sourceValue[1]}`);
+		assert(excludeResult2.affectedDependencies[0].included === false, 'Table t2 should be excluded as a result of excluding v1. Actual: true');
+
+		// including the view v1 should also include the table t2
+		const includeResult = await schemaCompareService.schemaCompareIncludeExcludeNode(operationId, v1Difference, true, azdata.TaskExecutionMode.execute);
+		this.assertIncludeExcludeResult(includeResult, true, 0, 1);
+		assert(includeResult.affectedDependencies[0].sourceValue[1] === 't2', `Table t2 should be the affected dependency. Actual: ${includeResult.affectedDependencies[0].sourceValue[1]}`);
+		assert(includeResult.affectedDependencies[0].included === true, 'Table t2 should be included as a result of including v1. Actual: false');
+
+		// excluding views from the comparison should make it so t2 can be excluded
+		deploymentOptions.excludeObjectTypes.push(mssql.SchemaObjectType.Views);
+		await schemaCompareService.schemaCompare(operationId, source, target, azdata.TaskExecutionMode.execute, deploymentOptions);
+		const excludeResult3 = await schemaCompareService.schemaCompareIncludeExcludeNode(operationId, t2Difference, false, azdata.TaskExecutionMode.execute);
+		this.assertIncludeExcludeResult(excludeResult3, true, 0, 0);
+	}
+
+	private assertIncludeExcludeResult(result: mssql.SchemaCompareIncludeExcludeResult, expectedSuccess: boolean, expectedBlockingDependenciesLength: number, expectedAffectedDependenciesLength: number): void {
+		assert(result.success === expectedSuccess, `Operation success should have been ${expectedSuccess}. Actual: ${result.success}`);
+		if (result.blockingDependencies) {
```

Comment:
I would say that if blockingDependencies/affectedDependencies is non-undefined then having expected lengths for these should be an error.

Reply:
Added error for these scenarios